### PR TITLE
Add shims for Apache Spark 3.2.1-SNAPSHOT builds

### LINKED ISSUE #3470: [FEA] Add shims for 3.2.1-SNAPSHOT
**Is your feature request related to a problem? Please describe.**
Apache Spark is now publishing nightly 3.2.1-SNAPSHOT builds from branch-3.2 and we need to add shims for this version.

**Describe the solution you'd like**
Add new shims.

**Describe alternatives you've considered**
None

**Additional context**
None

### PR DESCRIPTION
**Pull request by gerashegalov**
Closes #3470

Signed-off-by: Gera Shegalov <gera@apache.org>

### CODE REVIEW

**Comment by gerashegalov** [comment]
build

File: dist/pom.xml [hunk 1]
```diff
@@ -519,6 +519,15 @@
                                     <overWrite>false</overWrite>
                                     <outputDirectory>${project.build.directory}/deps</outputDirectory>
                                 </artifactItem>
+                                <artifactItem>
```
Comment (tgravescs) [inline dist/pom.xml:522]: assume this needs upmerge with pr to remove copy dependencies

**Comment by tgravescs** [comment]
looks good, other then assume one pr will need to go in first

**Comment by gerashegalov** [comment]
build

**Comment by pxLi** [comment]
> note we should update our nightly builds to test 3.2.1 too

I will enable the spark321 nightly pipeline

**Pull request merged**